### PR TITLE
Add install, embeddings enabled notices for post-hoc app setup

### PR DESCRIPTION
**Pull request by dominiccooney**
Adds popups to install app, set up embeddings, etc. when using the simplified onboarding flow.

Automatically switches to app for embeddings, when app has embeddings available.

Polishes vscode storybook so codicons work and there are more relevant style variables.

Part of #632.

Screenshot 2023-09-19 at 22 41 14

## Caveats

- The app prompts are still displayed on Windows, but there's no Windows app to download.
- [These prompts](https://www.figma.com/file/INGthyKM1cYki5H02JPDpW/VS-Code-Beta-2-%3C%3E-App-Onboarding?type=design&node-id=53-374&mode=design&t=ivzuiEzqtVsJtkD8-0) for Enterprise are not implemented. Enterprise uses the legacy widget.
- There's little/no logging for interactions with this UI.

## Test plan

```
pnpm -C vscode storybook
```

Open http://localhost:6007 and examine the App-less Onboarding storybooks.

Manual tests:

0. Uninstall Cody App. Remove test VScode state: `rm -rf /tmp/vscode-cody-extension-dev-host`
1. Run VScode with "Launch VScode Extension (Desktop; separate instance)" (equivalently, `--user-data-dir=/tmp/vscode-cody-extension-dev-host`) so you have no App token cached.
2. Open user settings JSON and add `"testing.simplified-onboarding": true` and restart VScode.
3. Create a new git repository with `git init /tmp/bar; cd /tmp/bar; git remote add origin git@host.example:bar/bar.git` . Open `/tmp/bar` in VScode.
4. Sign in with simplified onboarding.
5. The context underneath the chat box should be red. Click to open the notification. Check the link and Install App buttons work.
6. Install App. Click the Reload button. The prompt should change to Embeddings Not Found/Open App.
7. Click Open App, add embeddings for the repo. Wait for embeddings to complete. Back in VScode, click the Reload button. The indicator should change to a checked database icon. Or, if App presents a "back to VScode" button, click it and the indicator should change automatically. Or simulate that click by going `open vscode://sourcegraph.cody-ai/app-done' at the console.
8. Close app and wait 20 seconds. The indicator should go red/show a cross.
9. Open app and wait 20 seconds. The indicator should be checked.
10. Close app. Wait for the indicator to go read. Immediately open app, pop up the toast, and hit reload. The indicator should be checked immediately.
11. Clone something indexed by dotcom, for example, https://github.com/preactjs/preact . Open in VScode and open a file in the repository. The database icon should be checked (that is, using dotcom embeddings.)

### CODE REVIEW

**Review by toolmantim — COMMENTED** [review]
This is looking fantastic! And the storybook fixes are great too.

File: vscode/webviews/Popups/InstallCodyAppNotice.module.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    box-shadow: 0 0 8px 2px var(--vscode-widget-shadow);
+    border-radius: var(--notice-border-radius);
+
+    position: absolute;
+    bottom: calc(100% + var(--notice-border-radius));
+    left: calc(-1 * var(--notice-border-radius));
+    width: calc(100vh - 2rem);
+}
+
+.pointy-bit {
```
Comment (toolmantim) [inline vscode/webviews/Popups/InstallCodyAppNotice.module.css]: 🗡️

File: vscode/src/chat/ContextProvider.ts [hunk 2]
```diff
@@ -233,6 +242,48 @@ export class ContextProvider implements vscode.Disposable {
...
+
+    // Gets a list of GraphQL clients to interrogate for embeddings
+    // availability.
+    private async getEmbeddingClientCandidates(config: GraphQLAPIClientConfig): Promise<SourcegraphGraphQLAPIClient[]> {
+        const result = [new SourcegraphGraphQLAPIClient(config)]
+        if (isLocalApp(config.serverEndpoint)) {
+            // We will just talk to app.
+            return result
+        }
+        const appClient = await this.maybeAppClient()
```
Comment (thenamankumar) [inline vscode/src/chat/ContextProvider.ts:279]: What is the difference between App Client and GQL API Client?

And isLocalApp vs maybeAppClient?

File: vscode/webviews/ChatInputContextSimplified.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                        popupOpen={popupOpen}
+                        togglePopup={togglePopup}
+                        onboardingPopupProps={onboardingPopupProps}
+                    />
+                )
+            ) : (
+                // No codebase
+                <CodebaseState icon={mdiDatabaseOffOutline} />
+            )}
+            {(contextStatus?.filePath && (
```
Comment (thenamankumar) [inline vscode/webviews/ChatInputContextSimplified.tsx:82]: Better use `? :` and `&& ||`

File: vscode/webviews/Notices/EmbeddingsEnabledNotice.story.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    title: 'cody/App-less Onboarding',
+    component: EmbeddingsEnabledNotice,
+    decorators: [VSCodeStoryDecorator],
+}
+
+export default meta
+
+export const EmbeddingsEnabled: StoryObj<typeof EmbeddingsEnabledNotice> = {
+    render: () => (
+        <div style={{ background: 'rgb(28, 33, 40)' }}>
```
Comment (thenamankumar) [inline vscode/webviews/Notices/EmbeddingsEnabledNotice.story.tsx:17]: Probably use a class for styles?

File: vscode/webviews/Notices/EmbeddingsEnabledNotice.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { Notice } from './Notice'
+
+import styles from './EmbeddingsEnabledNotice.module.css'
+
+// A green shooting star. Hooray.
+const Icon: React.FunctionComponent = () => (
```
Comment (thenamankumar) [inline vscode/webviews/Notices/EmbeddingsEnabledNotice.tsx:6]: IF the Icon may be used in a different color in future, it is better to have `fill="currentColor` and then put `color: green` wherever we use the Icon. 

In that case, also better to extract Icon in a separate file.

File: vscode/webviews/Popups/Popup.module.css [hunk 6]
```diff
@@ -0,0 +1,90 @@
+.popup {
+    --notice-border-radius: 6px;
+    text-align: left;
+    box-sizing: border-box;
+    padding: 10px 10px 13px;
```
Comment (thenamankumar) [inline vscode/webviews/Popups/Popup.module.css:5]: I don't know about cody, but we don't do `px` in sourcegraph only `rem`.

**Review by thenamankumar — APPROVED** [review]

File: vscode/webviews/Notices/EmbeddingsEnabledNotice.tsx (hunk 5 above)
Comment (dominiccooney) [reply vscode/webviews/Notices/EmbeddingsEnabledNotice.tsx:6]: Thanks, TIL! I have done the currentColor refactoring and moved the color to the stylesheet.

I will leave the icon inline for now, we could reuse it by exporting the typescript symbol. The svg needs to be married to the style for the animation and that's how one more of these icons do it.

**Pull request merged**